packages/hash/api/src/graphql/resolvers/knowledge/page/set-parent-page.ts

```diff
@@ -0,0 +1,46 @@
+import { ApolloError } from "apollo-server-express";
+
+import { LoggedInGraphQLContext } from "../../../context";
+import {
+  MutationSetParentKnowledgePageArgs,
+  ResolverFn,
+} from "../../../apiTypes.gen";
+import {
+  mapPageModelToGQL,
+  UnresolvedKnowledgePageGQL,
+} from "../model-mapping";
+import { PageModel } from "../../../../model";
+
+export const setParentKnowledgePage: ResolverFn<
+  Promise<UnresolvedKnowledgePageGQL>,
+  {},
+  LoggedInGraphQLContext,
+  MutationSetParentKnowledgePageArgs
+> = async (
+  _,
+  { pageEntityId, parentPageEntityId, prevIndex = null, nextIndex = null },
+  { dataSources: { graphApi }, user },
+) => {
+  if (pageEntityId === parentPageEntityId) {
```

Comment:
This is interesting, do we stop cycles? Can I make `A` parent of `B` and `B` parent of `C` and `C` parent of `A`?

Reply:
These are the sorts of things we'd probably need to think about when it comes to authorization, because we can't have users messing up our invariants (if we can't enforce them on the Graph, which doesn't know about pages)